middleware.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Twitch Interactive, Inc.  All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not
+// use this file except in compliance with the License. A copy of the License is
+// located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package twirp
+
+import (
+	"context"
+)
+
+// Method is a method.
+type Method func(ctx context.Context, request interface{}) (interface{}, error)
+
+// Interceptor is a interceptor.
+type Interceptor func(Method) Method
+
+// ChainInterceptors chains the Interceptors.
+//
+// Returns nil if interceptors is empty.
```

Comment:
Let's remember to add more comments explaining `Method`, `Interceptor` and `ChainInterceptors`. The extended comments should contain some basic example. We should also add a new page for interceptors in the docs, and make sure it is linked from docs for [Hooks](https://twitchtv.github.io/twirp/docs/hooks.html), and from other pages that mention hooks.

Reply:
I added a bit more Godoc, and a skeleton for the `docs` folder - could you propose some more language you'd like to add here? Would be helpful.